parser/yaml_output_functions.c

```diff
@@ -0,0 +1,313 @@
+/***********************************************************************
+ *                   GNU Lesser General Public License
+ *
+ * This file is part of the GFDL Flexible Modeling System (FMS).
+ *
+ * FMS is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at
+ * your option) any later version.
+ *
+ * FMS is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+ **********************************************************************/
+#ifdef use_yaml
+
+#include <stdio.h>
+#include <yaml.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+
+struct fmsyamloutkeys {
+	char key1 [255];
+        char key2 [255];
+        char key3 [255];
+        char key4 [255];
+        char key5 [255];
+        char key6 [255];
+        char key7 [255];
+        char key8 [255];
+        char key9 [255];
+        char key10 [255];
+        char key11 [255];
+        char key12 [255];
+        char key13 [255];
+        char key14 [255];
+        char key15 [255];
```

Comment:
Is there a reason it was done this way with distinct key variables for each possible key? Would it better to just use a dynamic array typed like `char * keys[255]` or even statically with `char keys[15][255]` if we only need 15.

Reply:
There is a reason it's done this way.   This struct matches a C-bound fortran derived type.  My understanding is there there can be no pointers or allocatable memory in the struct/derived type for it to match.  I didn't want to create a 2D array because 2D arrays are arranged differently in C and Fortran, and the code would be difficult to maintain if I did that.  This was the best solution I could come up with.